### PR TITLE
Added ``about_point`` keyword argument to :class:`~.ApplyMatrix`

### PR DESCRIPTION
**Pull request by lanpai**
## Overview: What does this pull request change?

:class:`~ApplyMatrix` now has optional kwarg `about_point` in its `__init__` method.

## Motivation and Explanation: Why and how do your changes improve the library?

It allows for the animation to be much more robust and useful for different scenarios. (#2386)

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/animation/transform.py [hunk 1]
```diff
@@ -375,10 +375,14 @@ def create_target(self) -> Any:
 
 class ApplyMatrix(ApplyPointwiseFunction):
     def __init__(self, matrix: np.ndarray, mobject: Mobject, **kwargs) -> None:
+        if "about_point" not in kwargs:
+            kwargs["about_point"] = ORIGIN
+        about_point = kwargs["about_point"]
```
Comment (hydrobeam) [inline manim/animation/transform.py:380]: I feel like this may be better suited as a parameter.

**Review by hydrobeam — CHANGES_REQUESTED** [review]
Thank you for the contribution! I left a brief suggestion on its implementation.

Could you also add a brief description of the parameter ([see here for instructions](https://docs.manim.community/en/stable/contributing/docstrings.html)) and a quick test for it in `test_transform.py`? [See here](https://docs.manim.community/en/stable/contributing/testing.html) for how to add a test. Let me know if you have any questions 👍

**Comment by lanpai** [comment]
> Thank you for the contribution! I left a brief suggestion on its implementation.
> 
> Could you also add a brief description of the parameter ([see here for instructions](https://docs.manim.community/en/stable/contributing/docstrings.html)) and a quick test for it in `test_transform.py`? [See here](https://docs.manim.community/en/stable/contributing/testing.html) for how to add a test. Let me know if you have any questions 👍

`about_point` is now a parameter with relevant descriptions and tests implemented :+1:

**Review by hydrobeam — APPROVED** [review]
I added a quick unit test to compare it against `rotate`'s about_point, but LGTM!

File: tests/test_transform.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    square_2 = square_1.copy()
+
+    # rotation matrix
+    matrix = [[0, -1], [1, 0]]
+    about_point = np.asarray((-1.0, 0, 0.0))
+
+    ApplyMatrix(matrix, square_1, about_point=about_point)
+    square_2.rotate(90 * DEGREES, about_point=about_point)
+
+    np.testing.assert_array_equal(square_1.points.all() == square_2.points.all())
```
Comment (hydrobeam) [inline tests/test_transform.py:20]:
```suggestion
    np.testing.assert_array_equal(square_1.points, square_2.points)
```

File: manim/animation/transform.py [hunk 3]
```diff
@@ -374,11 +374,31 @@ def create_target(self) -> Any:
 
 
 class ApplyMatrix(ApplyPointwiseFunction):
-    def __init__(self, matrix: np.ndarray, mobject: Mobject, **kwargs) -> None:
+    """Applies a matrix transform to an mobject
```
Comment (behackl) [inline manim/animation/transform.py:377]:
```suggestion
    """Applies a matrix transform to an mobject.
```

File: manim/animation/transform.py [hunk 4]
```diff
@@ -374,11 +374,31 @@ def create_target(self) -> Any:
...
+
+    Parameters
+    ----------
+    matrix
+        The matrix transform
```
Comment (behackl) [inline manim/animation/transform.py:382]:
```suggestion
        The transformation matrix.
```

File: manim/animation/transform.py [hunk 5]
```diff
@@ -374,11 +374,31 @@ def create_target(self) -> Any:
...
+    mobject
+        The :class:`~.Mobject`.
+    about_point
+        The origin point for the transform. Defaults to ORIGIN
```
Comment (behackl) [inline manim/animation/transform.py:386]:
```suggestion
        The origin point for the transform. Defaults to ``ORIGIN``.
```

**Review by behackl — APPROVED** [review]
Thanks for your contribution, just added some minor docstring formatting suggestions. Will apply them myself and then merge this.

**Pull request merged**